Zend/zend_compile.c

```diff
@@ -12129,6 +12157,13 @@ static void zend_eval_const_expr(zend_ast **ast_ptr) /* {{{ */
 			return;
 		case ZEND_AST_CAST:
 			zend_eval_const_expr(&ast->child[0]);
+			if (ast->child[0]->kind == ZEND_AST_ZVAL) {
+				ZVAL_UNDEF(&result);
+				if (zend_try_ct_eval_cast(&result, ast->attr, zend_ast_get_zval(ast->child[0]))) {
+					break;
+				}
+				return;
```

Comment:
This return seems redundant, but idc too much about this if you prefer this style